core/server/common/src/main/java/alluxio/master/journal/raft/BufferedJournalApplier.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.raft;
+
+import alluxio.ProcessUtils;
+import alluxio.master.journal.AbstractCatchupThread;
+import alluxio.master.journal.CatchupFuture;
+import alluxio.master.journal.JournalEntryAssociation;
+import alluxio.master.journal.JournalUtils;
+import alluxio.master.journal.Journaled;
+import alluxio.master.journal.sink.JournalSink;
+import alluxio.proto.journal.Journal;
+import alluxio.resource.LockResource;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+/**
+ * Used to control applying to masters.
```

Comment:
Could you add some more info to the javadoc here (ie. we are applying journal entries and might need to buffer them if we are suspended)?